rero_mef/authorities/marctojson/helper.py

```diff
@@ -143,16 +162,21 @@ def build_string_list_from_fields(record, tag, subfields, separator):
     fields = record.get_fields(tag)
     field_string_list = []
     for field in fields:
-        subfield_list = []
+        subfield_string = ''
         for code, data in field:
-            if re.match('[' + subfields + ']', code):
-                subfield_list.append(data)
-        string_from_field = separator.join(subfield_list)
-        string_from_field = string_from_field.replace('\x98', '')
-        string_from_field = string_from_field.replace('\x9C', '')
-        string_from_field = string_from_field.replace(',,', ',')
-        string_from_field = re.sub(r'[,:;]$', '', string_from_field.strip())
-        field_string_list.append(string_from_field.strip())
+            if code in subfields:
+                if isinstance(data, (list, set)):
+                    data = subfields[code].join(data)
+                data = data.replace('\x98', '')
+                data = data.replace('\x9C', '')
+                data = data.replace(',,', ',')
+                data = remove_trailing_punctuation(data)
+                data = data.strip()
+                if subfield_string != '':
```

Comment:
as `data` is always a string, you can simply use `if subfield_string:`

Reply:
ok